lib/archethic/contracts/interpreter/library.ex

```diff
@@ -9,6 +9,66 @@ defmodule Archethic.Contracts.Interpreter.Library do
   alias Archethic.P2P.Message.GenesisAddress
   alias Archethic.P2P.Message.FirstPublicKey
 
+  @doc """
+  Returns the element at given index (zero-based)
+
+  ## Examples
+
+    iex> Library.at([1,2], 1)
+    2
+
+    iex> Library.at([], 0)
+    nil
+
+    iex> Library.at([], 0, "hello")
+    "hello"
+  """
+  def at(list, idx, default \\ nil) do
+    Enum.at(list, idx, default)
```

Comment:
Should we do a function for this or support list[i] ?

Reply:
TBH, I think we should all-in on the imperative way. Most people are used to it. Of course it's more work for the interpreter but that shouldn't be a concern.

loop: 
```
for (idx in list) do 

end
```

list access:
```
list[idx]
```

Reply:
I think the list access by index is indeed needed.

Reply:
After discussion and thinking, we might leverage a more functional approach. Just we need to ensure the naming of the function would be straightforward to understand without the knowledge of Elixir or functional programming